src/bulk/common.ts

```diff
@@ -167,59 +163,48 @@ export class Batch<T = Document> {
   }
 }
 
+function generateIdMap(ids: Document[]): { [key: number]: any } {
+  const idMap: { [index: number]: any } = {};
+  for (const doc of ids ?? []) {
```

Comment:
```suggestion
  for (const doc of ids) {
```
`ids` should always be defined as an array, according to Typescript

Reply:
Updated.